comet-java-client/src/main/java/ml/comet/experiment/impl/RestApiClient.java

```diff
@@ -0,0 +1,248 @@
+    }
+
+    Single<ExperimentAssetListResponse> getAssetList(String experimentKey, AssetType type) {
+        HashMap<QueryParamName, String> params = new HashMap<QueryParamName, String>() {{
+            put(EXPERIMENT_KEY, experimentKey);
+            put(TYPE, type.type());
+        }};
+        return singleFromSyncGet(GET_ASSET_INFO, params, ExperimentAssetListResponse.class);
+    }
+
+    Single<ExperimentStatusResponse> sendExperimentStatus(String experimentKey) {
+        return singleFromSyncGet(SET_EXPERIMENT_STATUS, experimentKey, ExperimentStatusResponse.class);
+    }
+
+    Single<LogDataResponse> logMetric(final MetricRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_METRIC, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logParameter(final ParameterRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_PARAMETER, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logOutputLine(final OutputUpdate request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_OUTPUT, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logHtml(final HtmlRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_HTML, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logOther(final LogOtherRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_LOG_OTHER, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> addTag(final AddExperimentTagsRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_TAG, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logGraph(final AddGraphRest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_GRAPH, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logStartEndTime(final ExperimentTimeRequest request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_START_END_TIME, LogDataResponse.class);
+    }
+
+    Single<LogDataResponse> logGitMetadata(final GitMetadata request, String experimentKey) {
+        request.setExperimentKey(experimentKey);
+        return singleFromAsyncPost(request, ADD_GIT_METADATA, LogDataResponse.class);
+    }
+
+    Single<CreateExperimentResponse> registerExperiment(final CreateExperimentRequest request) {
+        return singleFromSyncPost(request, NEW_EXPERIMENT, true, CreateExperimentResponse.class);
+    }
+
+    private <T> Single<T> singleFromAsyncPost(
```

Comment:
Some questions:
* Could we get a payload that is no encodable as JSON? If yes, what would be the behavior?
* What happens if the response is not a JSON?

Reply:
This method is intended only for JSON encoded payloads. I've not met non JSON encoded responses yet.
If not JSON encoded response received the error is going to be raised.
When we need to get non JSON response I'll implement corresponding method.

Reply:
We could get a non JSON-response in case of an HTTP error. I don't think it's likely we can get a non JSON-response with a status code in the [200, 300] range so it's ok for now.